core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -238,11 +256,51 @@ func (l *logPollerWrapper) LatestEvents() ([]evmRelayTypes.OracleRequest, []evmR
 			})
 		}
 	}
+	resultsReq = l.FilterPreviouslyDetectedRequests(resultsReq)
```

Comment:
We are risking being overloaded by always fully parsing everything and only later filtering old things out. It's super inefficient. Let's try to filter as early as possible, before all the heavy ABI decoding of commitments.